package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,337 @@
+    temp = aux.get_data("Temperature")
+    plt.plot(temp["Time"], temp["Temperature"])
+
+    # extract list of terms:
+    some_terms = aux.get_data(["Potential", "Kinetic En.", "Box-X"])
+
+    # extract all terms:
+    all_terms = aux.get_data()
+
+
+.. _EDR: https://manual.gromacs.org/current/reference-manual/file-formats.html#edr
+.. _XDR: https://datatracker.ietf.org/doc/html/rfc1014
+.. _pyedr: https://github.com/mdanalysis/panedr
+.. _GROMACS: https://github.com/gromacs/gromacs/blob/main/src/gromacs/fileio/enxio.cpp
+
+
+.. autoclass:: EDRReader
+   :members:
+
+
+The actual data for each step is stored by instances of EDRStep.
+
+.. autoclass:: EDRStep
+   :members:
+
+"""
+from pathlib import Path
+from typing import Optional, Union, Dict, List
+
+import numpy as np
+import pyedr
+
+from . import base
+
+
+class EDRStep(base.AuxStep):
+    """:class:`AuxStep` class for the .edr file format.
+
+    Extends the base AuxStep class to allow selection of time and
+    data-of-interest fields (by dictionary key) from the full set of data read
+    each step.
+
+    Parameters
+    ----------
+    time_selector : str, optional
+        Name of the dictionary key that links to the time values (assumed to
+        be in ps). Default value is "Time"
+    data_selector : str | list of str | None, optional
+        List of dictionary keys linking to data of interest in the EDR file to
+        be stored in ``data``. Default value is ``None``.
+    **kwargs
+        Other AuxStep options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxStep`
+    """
+
+    def __init__(self, time_selector: str = "Time",
+                 data_selector: Optional[str] = None, **kwargs):
+        super(EDRStep, self).__init__(time_selector=time_selector,
+                                      data_selector=data_selector,
+                                      **kwargs)
+
+    def _select_time(self, key: str) -> np.float64:
+        """'Time' is one of the entries in the dict returned by pyedr.
+        The base AuxStep Class uses the time_selector 'Time' to return the
+        time value of each step."""
+        return self._select_data(key)
+
+    def _select_data(self, key: Union[str, None]) -> np.float64:
+        if key is None:
+            return
+        try:
+            return self._data[key]
+        except KeyError:
+            raise KeyError(f"'{key}' is not a key in the data_dict dictionary."
+                           " Check the EDRReader.terms attribute")
+
+
+class EDRReader(base.AuxReader):
+    """ Auxiliary reader to read data from an .edr file.
+
+    Default reader for .edr files. All data from the file will be read and
+    stored on initialisation.
+
+    Parameters
+    ----------
+    filename : str
+        Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.
+
+    Attributes
+    ----------
+    _auxdata : pathlib.PosixPath
```

Comment:
A string linking to the file is turned into a PosixPath by pathlib's `Path.resolve()`, so _auxdata will not be of type string. This is changed from the XVGReader, where `os.path.abspath` was used instead of pathlib which does return a string